packages/x-date-pickers/src/TimeClock/ClockPointer.tsx

```diff
@@ -84,7 +84,7 @@ const ClockPointerThumb = styled('div', {
   boxSizing: 'content-box',
   variants: [
     {
-      props: { isBetweenTwoClockValues: false },
+      props: { isClockPointerBetweenTwoValues: false },
```

Comment:
This one fixes the unwanted white dot on the selected hour thumb. 🙈 
@siriwatknp Am I missing something or there is no way to infer `variants` props keys types? 🤔 
These regressions are annoying and really hard to spot. 🙈